calcom/cal.com
# New team creation - add Stripe checkout form

## What does this PR do?



This PR builds off of #5169. This PR adds a checkout step using Stripe elements to the new team creation wizard. 

Fixes #5166


https://www.loom.com/share/4cddc057ef12425384bba2074d5282e1

**Environment**: Staging(main branch) / Production

## Type of change





## How should this be tested?



When testing, ensure that you have the proper Stripe keys in your `.env`

Happy path
- Go to settings and choose "Add a team"
- Fill out the first page
- On the "Add team members page"
    - Invite a member via email
    - Invite a member via username
    - Invite that same member but search via email
    - Try inviting a previously invited email
        - Should not let you do this
    - Change the pricing from monthly and yearly
- When navigating to the checkout page check Stripe and see if the new customer is created with an inactive subscription
- Checkout with the success testing card
    - Check Stripe, the subscription should now be active
    - The new team should now be in the database
    - Team invite emails should be sent to those invited

Edge case - validating team name & URL
- Go back to the new team creation wizard
    - Try entering a team name and URL that is already in the database
        - Validation should stop you from doing this

Edge case - navigating between screens
- In the team creation wizard create a team and add data up to the purchase step
- Navigate to the previous pages, your data should be saved
- Change the number of members from the initial amount and go to the checkout page
    - On Stripe the number of seats of the inactive subscription should have updated

Edge case - exiting the new team wizard after Stripe customer has been made
- In the new team creation wizard create a team and navigate to the purchase step
- Close the window or navigate to another page
- Go back to the new team creation wizard and enter the same team name
    -  It should let you do this since the previous session was not written to the database
- Add a different amount of team members and go to the checkout page
    - On Stripe the customer that was created in the first session should now be updated

## Checklist


- I haven't added tests that prove my fix is effective or that my feature works
- I haven't checked if new and existing unit tests pass locally with my changes
